morango/api/permissions.py

```diff
@@ -33,7 +33,7 @@ def authenticate_credentials(self, userargs, password, request=None):
                 credentials[key] = val
 
         # authenticate the user via Django's auth backends
-        user = authenticate(**credentials)
+        user = authenticate(request=request, **credentials)
```

Comment:
Seemed like a good idea to make sure this passes along the request object, but it isn't required.

Reply:
Seems sensible, and it's not in the credentials dict, so seems low risk!